### PR TITLE
WIP: Fix 4259 - Update to Java 11

### PR DESCRIPTION
**Pull request by poikilotherm**
**Work in progress - do not merge - only for reference purposes.**

## Related Issues

- connects to #4259

## Tasks
- [ ] Evaluate possible problems with JDK9+ changes to [ByteBuffer](https://docs.oracle.com/javase/9/docs/api/java/nio/ByteBuffer.html) There have been a lot of changes to covariant methods, might be an issue for us.
- [ ] Fix all unit tests
- [ ] Enable integration testing with Java 11
- [ ] Documentation

### CODE REVIEW

**Comment by pdurbin** [comment]
@poikilotherm heads up that this pull request has merge conflicts.

**Comment by pdurbin** [comment]
@poikilotherm do we still need this pull request now that pull request #6320 has been merged? It has merge conflicts. Thanks.

**Comment by poikilotherm** [comment]
Not necessarily. Please keep in mind that this PR was about running Dataverse on Java 11, while #6320 is just about compilation for JDK 8 on a JDK 9+. You will still need to run this with JDK 8 (although I haven't tested what happens when running a JDK 8 WAR on a JDK 11).

**Comment by pdurbin** [comment]
@poikilotherm ok, but this pull request has merge conflicts. And it's almost our oldest pull request. Should we close it?

**Comment by poikilotherm** [comment]
Up to you. I let this thing stall as we had no movement on the appserver front, which is most likely to be solved first. There is no real pressure on getting of Java 8 right now.

File: pom.xml [hunk 1]
```diff
@@ -680,5 +680,37 @@
         <id>all-unit-tests</id>
       </profile>
       <!-- TODO: Add a profile to run API tests (integration tests that end in IT.java. See conf/docker-aio/run-test-suite.sh -->
+        <profile>
+            <id>java11</id>
+            <properties>
+                <java.version>11</java.version>
+                <maven.compiler.source>${java.version}</maven.compiler.source>
+                <maven.compiler.target>${java.version}</maven.compiler.target>
```
Comment (pdurbin) [inline pom.xml]: Are "source" and "target" deprecated in Java 9 and later? That's what pull request #6320 was about, I thought.

**Pull request closed without merging**